File: project/notifications/backend/services/notifications-manager-service/src/main/java/com/jaruiz/casarrubios/notifications/services/api/async/scoring/dto/ApplicationScoredEventDTO.java
package com.jaruiz.casarrubios.notifications.services.api.async.scoring.dto;

import java.math.BigDecimal;
import java.util.List;

import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@NoArgsConstructor
public class ApplicationScoredEventDTO {
    private String applicationId;
    private int positionId;
    private ResumeAnalysisDTO analysis;
    private ScoringDTO scoring;

    @Data
    @NoArgsConstructor
    public static class ResumeAnalysisDTO {

        private String summary;
        private List<String> strengths;
        private List<String> concerns;
        private List<SkillDTO> hardSkills;
        private List<SkillDTO> softSkills;
        private List<String> keyResponsibilities;
        private List<String> interviewQuestions;
        private int totalYearsExperience;
        private BigDecimal averagePermanency;
        private List<String> tags;
    }

    @Data
    @NoArgsConstructor
    public static class ScoringDTO {
        private BigDecimal score;
        private BigDecimal descScore;
        private BigDecimal requirementScore;
        private BigDecimal tasksScore;
        private BigDecimal timeSpent;
        private String explanation;
    }

    @Data
    @NoArgsConstructor
    public static class SkillDTO {
        private String skill;
        private int level;
    }
}

File: project/recruiters/backend/services/applications-service/src/api/input/message/dto/application_scored_event_dto.py
from dataclasses import dataclass
from typing import List


@dataclass
class SkillDTO:
    skill: str
    level: str

    def __getitem__(self, item):
        return self[item]

@dataclass
class ResumeAnalysisDTO:
    summary: str
    strengths: List[str]
    concerns: List[str]
    hardSkills: List[SkillDTO]
    softSkills: List[SkillDTO]
    keyResponsibilities: List[str]
    interviewQuestions: List[str]
    totalYearsExperience: int
    averagePermanency: float
    tags: List[str]

    def __getitem__(self, item):
        return self[item]

@dataclass
class ScoringDTO:
    score: float
    descScore: float
    requirementScore: float
    tasksScore: float
    timeSpent: float
    explanation: str

    def __getitem__(self, item):
        return self[item]

@dataclass
class ApplicationScoredEventDTO:
    applicationId: str
    candidateId: str
    positionId: int
    analysis: ResumeAnalysisDTO
    scoring: ScoringDTO

    def __getitem__(self, item):
        return self[item]
